src/math.cpp

```diff
@@ -304,41 +455,62 @@ auto modf(const CallContext& ctx) -> Vallist {
 }
 
 auto rad(const CallContext& ctx) -> Value {
+    auto origin = Origin(UnaryOrigin{
+        .val = make_owning<Value>(ctx.arguments().get(0)),
+        .location = ctx.call_location(),
+        .reverse = [](const Value& new_value,
+                      const Value& old_value) -> std::optional<SourceChangeTree> {
+            Number n = std::get<Number>(new_value);
+            return old_value.force(n * 180 / PI);
+        }});
+
     return math_helper(
-        ctx, [](double d) { return d * PI / 180; }, "rad");
+               ctx, [](double d) { return d * PI / 180; }, "rad")
+        .with_origin(origin);
 }
 
 auto random(const CallContext& ctx) -> Value {
+    auto origin = Origin(BinaryOrigin{
+        .lhs = make_owning<Value>(ctx.arguments().get(0)),
+        .rhs = make_owning<Value>(ctx.arguments().get(1)),
+        .location = ctx.call_location(),
+        .reverse = [](const Value& /*unused*/, const Value& /*unused*/, const Value& /*unused*/)
+            -> std::optional<SourceChangeTree> { return std::nullopt; } // TODO: return it correctly
```

Comment:
A random value can't be forced. So returning `nullopt` is fine. But we should note that in a comment.

Also if you want to just always prevent forcing, it's enough to just remove the origin from the returned value. You don't **need* to add an origin.